### PR TITLE
bpo-35717: Fix KeyError exception raised when using enums and compile

### PR DESCRIPTION
**Pull request by remilapeyre**
https://bugs.python.org/issue17467

### CODE REVIEW

File: Lib/test/test_enum.py [hunk 1]
```diff
@@ -1858,6 +1858,13 @@ class Decision2(MyEnum):
             REVERT_ALL = "REVERT_ALL"
             RETRY = "RETRY"
 
+    def test_compile(self):
```
Comment (vstinner) [inline Lib/test/test_enum.py]: Please add a short comment mentioning bpo-35717 to explain the rationale of this test.

Comment (serhiy-storchaka) [reply Lib/test/test_enum.py]: Decorate the test with `cpython_only`. And I think it needs better name. It does not have particular relation with compiling. Maybe "test_empty_globals"?

Comment (serhiy-storchaka) [reply Lib/test/test_enum.py]: Hmm, no, decorating with `cpython_only` is not needed. The test should pass on other implementations even if they do not have `_getframe()`.

Comment (ethanfurman) [reply Lib/test/test_enum.py]: Add the comment @vstinner suggested and you're good to go.

**Review by ethanfurman — CHANGES_REQUESTED** [review]

File: Lib/test/test_enum.py (hunk 1 above)
Comment (remilapeyre) [reply Lib/test/test_enum.py]: Thanks, I added the comment

**Comment by remilapeyre** [comment]
I have made the requested changes; please review again

**Review by ethanfurman — CHANGES_REQUESTED** [review]
Looks good.

Now add yourself to `Misc/ACKS`.  :)

**Comment by remilapeyre** [comment]
Thanks @ethanfurman, I have made the requested changes; please review again.

**Comment by ethanfurman** [comment]
Thank you.  As soon as I get this on my machine for final checks I'll move it forward.  :)

**Comment by vstinner** [comment]
> Now add yourself to Misc/ACKS. :)

Honestly, I don't think that it it's still important since we moved to Git: the author is know tracked in Git.

Note: There is a "thanks.python.org" project similar to https://thanks.rust-lang.org/ cc @pablogsal

**Comment by pablogsal** [comment]
> Note: There is a "thanks.python.org" project similar to https://thanks.rust-lang.org/ cc @pablogsal

Sadly, it was cancelled as in the core dev sprint it was decided not worth the maintenance cost.

**Review by ethanfurman — APPROVED** [review]

**Pull request merged**